lua/luasnip/config.lua

```diff
@@ -188,10 +188,16 @@ c = {
 			"Remove buffers' nodes on deletion+wipeout.
 			autocmd BufDelete,BufWipeout * lua current_nodes = require("luasnip").session.current_nodes if current_nodes then current_nodes[tonumber(vim.fn.expand("<abuf>"))] = nil end
 		]]
-					.. (session.config.enable_autosnippets and [[
+					.. (
+						session.config.enable_autosnippets
+							-- Luasnip_just_inserted makes sure we only trigger autosnippets after we inserted characters in insert mode.
+							-- vim.b.Luasnip_last_changedtick allows us to compute how many changes occurred in the file between two TextChanged* events. We use this information to distinguish typing characters by hand vs using <C-r> to paste multiple characters in insert mode. We want to allow expanding autosnippets in the first case and not in the second. After some investigation, this is the only way I found to distinguish the two cases.
```

Comment:
I'm not 100% happy with this comment, I think it could be more descriptive, ie:
1. what is the issue
2. how can typing characters and pasting multiple characters be distinguished
3. the solution

Also, could you make the comment 80-100 characters wide? Normally stylua should take care of this, but here it doesn't :shrug:

Reply:
(Actually, maybe stylua isn't supposed to..)

Reply:
Oh, and in a similar vein, could you shorten the comments in the tests?